### ISSUE TITLE
Crash if a TypeScript class defines a `request` method

### ISSUE THREAD

**Issue by topiolli** [mozilla/sphinx-js#170]
~~~ ts
export default class Test {
  private request(): void {
  }
}
~~~

~~~
Traceback (most recent call last):
  File "/home/topi/code/sphinx/sphinx/events.py", line 111, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/topi/code/sphinx-js/sphinx_js/__init__.py", line 60, in analyze
    app._sphinxjs_analyzer = analyzer.from_disk(abs_source_paths,
  File "/home/topi/code/sphinx-js/sphinx_js/typedoc.py", line 40, in from_disk
    return cls(json, base_dir)
  File "/home/topi/code/sphinx-js/sphinx_js/typedoc.py", line 33, in __init__
    self._objects_by_path.add_many((obj.path.segments, obj) for obj in ir_objects)
  File "/home/topi/code/sphinx-js/sphinx_js/suffix_tree.py", line 46, in add_many
    self.add(segs, value)
  File "/home/topi/code/sphinx-js/sphinx_js/suffix_tree.py", line 31, in add
    tree[seg] = Value(value)
TypeError: 'Value' object does not support item assignment
~~~

Each node (`dict`) in `suffix_tree` always seems to contain a "request" key as its first entry. Having a method with the same name breaks things somehow.

**Comment by erikrose** [comment]
Great reduction; thanks!

**Comment by erikrose** [comment]
I threw together a quick test, and, surprisingly, it passed:

```python
class Bug170Tests(TypeDocAnalyzerTestCase):
    """Make sure ``make_path_segments() `` works on all its manifold cases."""

    files = ['bug170.ts']

    def test_crash(self):
        """Make sure it doesn't crash."""
        obj = self.analyzer.get_object(['Test'])
```

`obj` ends up being...

```python
Class(members=[Function(is_abstract=False, is_optional=False, is_static=False, is_private=True, name='request', path=<Pathname(['./', 'bug170.', 'Test#', 'request'])>, filename='bug170.ts', description='', line=2, deprecated=False, examples=[], see_alsos=[], properties=[], exported_from=None, params=[], exceptions=[], returns=[])], supers=[], name='Test', path=<Pathname(['./', 'bug170.', 'Test'])>, filename='bug170.ts', description='', line=1, deprecated=False, examples=[], see_alsos=[], properties=[], exported_from=<Pathname(['./', 'bug170'])>, constructor=None, is_abstract=False, interfaces=[])
```

Are you up to date? I did fix a SuffixTree bug a little bit ago.

**Comment by erikrose** [comment]
I just found 2 other SuffixTree bug repros floating around on a personal branch. Let me see if I can fix those. Maybe they will make your problem evaporate.

**Comment by erikrose** [comment]
Fixed those bugs and more in e16789fbe5a9edc6eafb96fafdf2929d2603825b. Does that solve your problem?

**Comment by willkg** [comment]
Given there's no response since 2021, I'm going to assume this is solved now.
